### PR TITLE
[WIP] Update projects to dotnet 1.0.1

### PR DESCRIPTION
**Pull request by ncave**
@enricosada Do you think there is a chance to get another release of this (with or without this PR)? It will help keep FCS going until the new .fsproj is supported by VS2017.

### CODE REVIEW

**Comment by cloudRoutine** [comment]
@ncave I can do another one for you, once you've fixed the CI failures

File: src/dotnet-fssrgen/project.json [hunk 1]
```diff
@@ -1,51 +0,0 @@
-{
-    "title": "F# String Response Generator - dotnet cli tool",    
-    "version": "3.3.0",
-    "authors": ["Don Syme","Enrico Sada","Jared Hester"],
```
Comment (cloudRoutine) [inline src/dotnet-fssrgen/project.json:4]: you need to move these too, they'll be added to the `.nuspec` generated by `dotnet pack`

File: global.json [hunk 2]
```diff
@@ -1,4 +0,0 @@
-{
-  "projects": [ "src", "test" ]
```
Comment (cloudRoutine) [inline global.json]: don't delete this file, specify the dotnet cli version in it

File: src/dotnet-fssrgen/dotnet-fssrgen.fsproj [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="FSharp.NET.Sdk;Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
```
Comment (cloudRoutine) [inline src/dotnet-fssrgen/dotnet-fssrgen.fsproj]: should be updated to `netcoreapp1.1` since it supports 12 more platforms

**Comment by ncave** [comment]
@cloudRoutine Can you do one without this PR, that works too. Thanks a lot!

File: src/dotnet-fssrgen/dotnet-fssrgen.fsproj [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="../FsSrGen/Program.fs" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="System.Xml.XmlDocument" Version="4.3.0" />
```
Comment (cloudRoutine) [inline src/dotnet-fssrgen/dotnet-fssrgen.fsproj]: awww crap, this means I gotta go rewrite some code to use `XDocument` instead 😑

**Review by cloudRoutine — CHANGES_REQUESTED** [review]

**Comment by cloudRoutine** [comment]
If I'm going to do a release, I'd rather not do another one right after to include stuff that should have gone into the first one.

File: src/dotnet-fssrgen/project.json (hunk 1 above)
Comment (ncave) [reply src/dotnet-fssrgen/project.json:4]: Added.

**Comment by cloudRoutine** [comment]
you should migrate the other projects too, then I can do a `3.4.0` release for all of them at once. 
plus this is probably the source of the CI failures 😉

**Comment by enricosada** [comment]
> Do you think there is a chance to get another release of this (with or without this PR)? It will help keep FCS going until the new .fsproj is supported by VS2017

i ❤️ this PR. and thanks a lot!
but why is needed for FCS? `dotnet-fssrgen` is a cli tool, target should not matter (maybe i am missing something)

**Comment by ncave** [comment]
@enricosada Perhaps I did not phrase it well, what I tried to say is that FCS dotnet core build target still uses the dotnet-fssrgen cli. The current version on nuget (3.3.0) does not contain the latest master, so I was wondering if it is possible to release another that has the latest master. This is not really related to this PR which only does the upgrade to 1.0.1, but perhaps having it run on latest dotnet core will be a better release.

**Comment by enricosada** [comment]
ah ok, sure, this is so near, let's just release after this is merged

**Comment by ncave** [comment]
@cloudRoutine @enricosada
Current state:
- updated all projects (except the test for msbuild task, as the task is already handled by Don, I think)
- packages are successfully created (build CreatePackages)
- for some reason can't add net46 as target
- for some reason can't restore the local package in test/use-dotnet-fssrgen-as-tool

**Comment by ncave** [comment]
@enricosada

- updated all projects (except the test project for msbuild task, because of #10)
- packages are successfully created (build CreatePackages works)
- tests are now running (build RunTests) but failing to find the resource:
`System.Resources.MissingManifestResourceException : Could not find any resources appropriate for the specified culture or the neutral culture.`

**Comment by ncave** [comment]
@cloudRoutine @enricosada 
- updated all projects (except the test project for msbuild task, because of #10)
- packages are successfully created (build CreatePackages works)
- tests are now passing (build RunTests) (test for msbuild task is disabled)

Can you please review, thanks.

**Comment by cloudRoutine** [comment]
95-102 in build.fsx are using outdated dotnet cli commands, this is what's causing travis to fail

**Review by cloudRoutine — APPROVED** [review]

**Comment by cloudRoutine** [comment]
Other than the last few build config hiccups to straighten out, LGTM.
I'll release a new version once they're both passing. thanks @ncave

**Comment by ncave** [comment]
@cloudRoutine 
- The only way I could make travis CI pass was to remove the net46 target, please let me know if it's ok.
- If not, and you're not packaging from Travis CI, perhaps we can replace the "dotnet pack" in travis.yml with "dotnet build --framework netcoreapp1.1"?
- Also, osx keeps failing for various reasons, I made it optional in the matrix.

**Comment by cloudRoutine** [comment]
@ncave the latest versions are up

**Comment by ncave** [comment]
@cloudRoutine Thanks!

**Pull request merged**